Assets/Scripts/Gameplay/UI/UI.cs

```diff
@@ -0,0 +1,21 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class UI : MonoBehaviour
+{
+    public GameObject MessagePanel;
+    public Text MessageText;
+
+    public void OpenMessagePanel(string text = null) {
+        if (text != null) {
+            MessageText.text = text;
```

Comment:
Use `this` to access members. (there are many other places in this PR where you need to do that as well)